### DISCUSSION TITLE
How to utilize jQuery in Commerce SDK  project?

### DISCUSSION THREAD

**Discussion by zhangguanghuib** [microsoft/Dynamics365Commerce.Solutions#241] [Q&A] [answered]
Partner develop is asking how to utilize jQuery in Commerce SDK  POS extension project,  I tried some ways but none of them working,  can you please advise how to utilize jQuery?<br/>
What I did is:<br/>
- Install Jquery Nuget Package to the POS  extension project:
<img width="1462" alt="image" src="https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/14832260/7eebeb89-e605-4afe-9bc8-070d0e37714b">

- Manifest file:
<img width="1297" alt="image" src="https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/14832260/ea954674-ae1e-4f3f-8c34-32a769f81a47">

- tsconfig file:
<img width="532" alt="image" src="https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/14832260/6d3d0ecc-80f3-4327-9629-8efe80da007d">

But the problem is:<br/>

In the extension code, when import $ and then try use it:<br/>
<img width="954" alt="image" src="https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/14832260/df5b2741-a2d1-47d8-ada8-1bf83c29c940">

When open Store Commerce, we will face this application error:
<img width="251" alt="image" src="https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/14832260/13654227-35a3-43c4-a0d6-332b8b068035">

Question:

Can you please check how jQuery is able to be utilized in Commerce SDK  POS  extension project?

**Comment by madyke** [comment]
Hello @zhangguanghuib - if you haven't seen it already, there's a doc to help get third-party libraries like jQuery set up:
[Consume external or third party libraries like Knockout.js in POS extensions](https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/pos-extension/knockout-pos-extension)
It looks like you've got everything set up correctly, but if you haven't seen this doc it's probably worth double checking to make sure you followed all the steps there.

Assuming all of the setup is correct, you'll probably need to take a look at the error that Store Commerce is throwing when you open it. It may actually be an issue with whatever the `.tooltip` function is that it's complaining about, rather than a problem loading jQuery itself. You should be able to put a breakpoint there or inspect the error. If the `$` object exists (meaning it's not null) then jQuery is loading correctly and the error is actually with some implementation in the code.

**Comment by zhangguanghuib** [comment, marked as answer]
@madyke I fixed this issue with these steps:
1. Download jQuery.d.ts from https://www.nuget.org/packages/jquery.TypeScript.DefinitelyTyped/ <br/>
2. Put in under any folder like this: <br/>
<img width="1301" alt="image" src="https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/14832260/bf44b1f8-6fc8-4ae2-9c95-d4d5efca9f32"><br/>
Then it worked
